Dockerfile

```diff
@@ -22,29 +22,48 @@ ARG SKIP_MINIFY
 
 EXPOSE 8888
 
+# Set the default environment to be CI, can be overridden by passing new CONFIG_ENV setting at container start
+ENV CONFIG_ENV ci
+ENV DOCKER_CONTAINER true
+
 # Copy in the narrative repo
-ADD ./ /kb/dev_container/narrative
 ADD ./kbase-logdb.conf /tmp/kbase-logdb.conf
+ADD ./narrative-log-proxy.conf /kb/dev_container/narrative/
 ADD ./deployment/ /kb/deployment/
-WORKDIR /kb/dev_container/narrative
+ADD ./build.js /kb/dev_container/narrative/
+ADD ./Gruntfile.js /kb/dev_container/narrative/
+ADD ./Makefile /kb/dev_container/narrative/
+ADD ./scripts /kb/dev_container/narrative/scripts
+ADD ./nbextensions /kb/dev_container/narrative/nbextensions
+# JS prereqs
+ADD ./package.json /kb/dev_container/narrative/
+ADD ./package-lock.json /kb/dev_container/narrative/
+# JS tests
+ADD ./test /kb/dev_container/narrative/test
+# all the UI stuff
+ADD ./kbase-extension /kb/dev_container/narrative/kbase-extension
 
-# Core Python necessities + KBase scientific computing packages
-RUN pip install -r ./docker_image_dependencies/python-requirements.txt
+# narrative backend
+ADD ./src /kb/dev_container/narrative/src
+# needed for narrative_version
+ADD ./.git /kb/dev_container/narrative/.git
```

Comment:
Copying over the directories and files that are actually used, instead of everything under the sun. Attempted to order them by approximate likelihood of being changed, although everything after `nbextensions` is a lot more likely to be edited than the stuff before it.